cmd/initialise.go

```diff
@@ -0,0 +1,189 @@
+
+package cmd
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"github.com/spf13/cobra"
+)
+
+var httpClient    http.Client
+
+// InitPayload holds a Vault init request.
+type InitPayload struct {
+	RecoveryShares    int `json:"recovery_shares"`
+	RecoveryThreshold int `json:"recovery_threshold"`
+	SecretShares      int `json:"secret_shares"`
+	SecretThreshold   int `json:"secret_threshold"`
+}
+
+// InitResponse holds a Vault init response.
+type InitResponse struct {
+	Keys               []string `json:"keys"`
+	KeysBase64         []string `json:"keys_base64"`
+	RootToken          string   `json:"root_token"`
+	RecoveryKeys       []string `json:"recovery_keys"`
+	RecoveryKeysBase64 []string `json:"recovery_keys_base64"`
+}
+
+// initialiseCmd represents the initialise command
+var initialiseCmd = &cobra.Command{
+	Use:   "initialise",
+	Short: "A brief description of your command",
+	Long: `A longer description that spans multiple lines and likely contains examples
+and usage of using your command. For example:
+
+Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		// make vault init request and get root token
+		fmt.Println("Initialising Vault...")
+		rootToken := initVault()
+		fmt.Println("Initialisation complete")
+
+		// AWS setup
+		region := "us-east-1"
+		sess := session.Must(session.NewSession(&aws.Config{Region: aws.String(region)}))
+		var kmsClient = kms.New(sess, aws.NewConfig().WithRegion(region))
+		uploader := s3manager.NewUploader(sess)
+
+		// encrypt tokens with AWS KMS
+		fmt.Println("Encrypting root token...")
+		encryptedToken, errE := kmsClient.Encrypt(&kms.EncryptInput{
+			KeyId: aws.String(fullKeyID(os.Args[1], os.Args[2])),
+			Plaintext: []byte(rootToken),
+		})
+		checkError(errE)
+		fmt.Println("Encryption complete.")
+
+		hostname, errH := os.Hostname()
+		if errH != nil {
+			panic(errH)
+		}
+
+		tokenFileName := hostname+ "_token"
+
+		writeToFile(tokenFileName, encryptedToken.CiphertextBlob)
+
+		// upload keys to S3
+		fmt.Println("Uploading encrypted token to S3...")
+		f := openFile(tokenFileName)
+
+		s3Result, errS3 := uploader.Upload(&s3manager.UploadInput{
+			Bucket: aws.String("encrypted-tokens"),
+			Key:    aws.String(tokenFileName),
+			Body:   f,
+		})
+		checkError(errS3)
+		fmt.Println("Encrypted token successfully uploaded to S3 at", s3Result.Location)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(initialiseCmd)
+}
+
+func checkError(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+func initVault() string {
+	vaultAddr := os.Getenv("VAULT_ADDR")
+	if vaultAddr == "" {
+		vaultAddr = "https://127.0.0.1:8200"
```

Comment:
For local, the default should be `http` since we don't have certificates locally, right?

```suggestion
		vaultAddr = "http://127.0.0.1:8200"
```